src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.cs

```diff
@@ -332,10 +332,12 @@ private void SetErrorResponseHeaders(int statusCode)
         public unsafe void SendResponseHeaders(bool appCompleted)
         {
             // Verifies we have sent the statuscode before writing a header
-            var reasonPhraseBytes = Encoding.UTF8.GetBytes(ReasonPhrase ?? ReasonPhrases.GetReasonPhrase(StatusCode));
+            var reasonPhrase = string.IsNullOrWhiteSpace(ReasonPhrase) ? ReasonPhrases.GetReasonPhrase(StatusCode) : ReasonPhrase;
+            var reasonPhraseBytes = Encoding.UTF8.GetBytes(reasonPhrase);
```

Comment:
If I set a status code that had an undefined reason phrase would I have the same issue? E.g. GetReasonPhrase can return null or empty (?)